client/maps/action-creators.ts

```diff
@@ -0,0 +1,221 @@
+import {
+  GetMapDetailsPayload,
+  GetMapsPayload,
+  MapInfoJson,
+  MapPreferences,
+  MapSortType,
+  MapVisibility,
+  Tileset,
+  UpdateMapPayload,
+  UpdateMapServerBody,
+} from '../../common/maps'
+import { ThunkAction } from '../dispatch-registry'
+import fetch from '../network/fetch'
+import { apiUrl } from '../network/urls'
+import { openSnackbar } from '../snackbars/action-creators'
+import { ClearMaps } from './actions'
+
+const upload = IS_ELECTRON ? require('./upload').default : null
+
+export function uploadLocalMap(path: string, onMapSelect: (map: MapInfoJson) => void): ThunkAction {
+  return async dispatch => {
+    if (!upload) {
+      return
+    }
+
+    dispatch({
+      type: '@maps/uploadLocalMapBegin',
+      payload: { path },
+    })
+
+    dispatch({
+      type: '@maps/uploadLocalMap',
+      payload: upload(path, apiUrl`maps`).then(({ map }: { map: MapInfoJson }) => {
+        if (onMapSelect) {
+          onMapSelect(map)
```

Comment:
This is a bad place to do this because if `onMapSelect` throws, it will turn the action into a failure (even though it succeeded and the map is now on the server). I would instead pull the `upload` call out of the dispatch call and store the promise, pass it into the dispatch, and attach a `then` to handle this call that won't affect the result given to the reducers